background_scripts/main.js

```diff
@@ -182,6 +182,67 @@ const mkRepeatCommand = (command) => (function (request) {
   }
 });
 
+function nextZoomLevel(currentZoom, steps) {
+  const chromeLevels = [ // Chrome's default zoom levels.
+    0.25,
+    0.33,
+    0.50,
+    0.75,
+    0.80,
+    0.90,
+    1.00,
+    1.10,
+    1.25,
+    1.50,
+    1.75,
+    2.00,
+    2.50,
+    3.00,
+    4.00,
+    5.00,
+  ];
+  const firefoxLevels = [ // Firefox's default zoom levels.
+    0.30,
+    0.50,
+    0.67,
+    0.80,
+    0.90,
+    1.00,
+    1.20,
+    1.33,
+    1.50,
+    1.70,
+    2.00,
+    2.40,
+    3.00,
+    4.00,
+    5.00,
+  ];
+
+  let zoomLevels = chromeLevels; // Chrome by default
+  if (BgUtils.isFirefox()) {
+    zoomLevels = firefoxLevels;
+  }
+
+  if (steps === 0) { // Nothing
+    return currentZoom;
+  } else if (steps > 0) { // In
+    // Chrome sometimes returns values with floating point errors.
+    currentZoom += 0.0000001; // This is needed to solve floating point bugs in Chrome.
```

Comment:
We need to see if adding a tiny value will boost the number up to a value in our list to cover floating point errors.

Reply:
What does floating point error mean here?  isNaN?

Reply:
I add an example into the comments to make this meaning clear. There are also unit tests to cover the known values for which Chrome does not return the precise number.
For further clarification, if we run `tabs.setZoom(tabId, 0.33)` Chrome will show 33% in the UI. However, when we run `tabs.getZoom(tabId)` Chrome will give us the number `0.32999999999999996`